packages/runtime/container-runtime/src/pendingStateManager.ts

```diff
@@ -306,6 +306,27 @@ export class PendingStateManager implements IDisposable {
 		message: InboundSequencedContainerRuntimeMessage;
 		localOpMetadata: unknown;
 	}[] {
+		if (batch.length === 0) {
+			const pendingMessage = this.pendingMessages.peekFront();
+			assert(pendingMessage !== undefined, "No pending message found for this remote message");
+			if (pendingMessage.batchIdContext.batchStartCsn !== batchStartCsn) {
+				this.stateHandler.close(
+					DataProcessingError.create(
+						"batchStartCsn mismatch on empty batch",
+						"unexpectedAckReceived",
+						undefined,
+						{
+							expectedMessageType: JSON.parse(pendingMessage.content).type,
+						},
+					),
+				);
+				return [];
+			}
+			this.savedOps.push(withoutLocalOpMetadata(pendingMessage));
```

Comment:
Are you sure we want to touch savedOps here? What will that do to add this empty groupedBatch op?